### PR TITLE
WINDUPRULE-353 Set of rules to cover potential issues when migrating from oracle to openjdk

### PR DESCRIPTION
**Pull request by jeichler**
https://issues.jboss.org/browse/WINDUPRULE-353

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by mrizzi** [comment]
retest

**Comment by jeichler** [comment]
@mrizzi ack. will take those comments into consideration for the next submission of rules.
thanks for making the changes!

File: .gitignore [hunk 1]
```diff
@@ -29,4 +29,4 @@ rules-reviewed/CreateIDs.xsl
 rules-reviewed/CreateTests.xsl
 rules-reviewed/tags.txt
 tags.txt
-
+.history
```
[the remark this answers is not in the document]
Comment (jeichler) [reply .gitignore]: done

File: rules-reviewed/openjdk/oracle2openjdk.rhamt.xml [hunk 2]
```diff
@@ -0,0 +1,160 @@
+<?xml version="1.0"?>
```
[the remark this answers is not in the document]
Comment (jeichler) [reply rules-reviewed/openjdk/oracle2openjdk.rhamt.xml]: done

File: rules-reviewed/openjdk/oracle2openjdk.rhamt.xml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            <perform>
+                <hint title="Crypto: elliptic curves usage" effort="0" category-id="potential">
+                    <message>
+                    When on RHEL, OpenJDK uses the NSS crypto library from RHEL instead of the built-in one.
+
+                    This mostly affects elliptic curve cryptography, where OpenJDK supports all of the RHEL curves and none of the others.
+                    
+                    A list of all cipher suits of NSS in RHEL:
+
+                    * RHEL 6: https://access.redhat.com/articles/1470663
```
[the remark this answers is not in the document]
Comment (jeichler) [reply rules-reviewed/openjdk/oracle2openjdk.rhamt.xml]: done

**Comment by mrizzi** [comment]
retest

File: rules-reviewed/openjdk/tests/oracle2openjdk.rhamt.test.xml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <testDataPath>data</testDataPath>
+    <rulePath>../oracle2openjdk.rhamt.xml</rulePath>
+    <sourceMode>true</sourceMode>
+    <ruleset>
+        <rules>
+            <rule id="oracle2openjdk-00000-test">
+                <when>
+                    <not>
+                        <hint-exists message="Currently OpenJFX"/>
```
[the remark this answers is not in the document]
Comment (jeichler) [reply rules-reviewed/openjdk/tests/oracle2openjdk.rhamt.test.xml]: done

**Comment by jeichler** [comment]
@mrizzi f*** seems I messed up during the rebase. another PR incoming later, correcting this mess.

**Comment by jeichler** [comment]
continued here: https://github.com/windup/windup-rulesets/pull/342

**Pull request closed without merging**